src/Catel.Core/Catel.Core.Shared/Data/ModelBase.cs

```diff
@@ -312,6 +313,19 @@ protected internal bool LeanAndMeanModel
         public static bool DefaultDisableEventSubscriptionsOfChildValuesValue { get; set; }
 
         /// <summary>
+        /// Gets or sets a value indicating whether usage of equality comparer for equals should be disabled.
+        /// <para />
+        /// The default value is <c>false</c>.
+        /// </summary>
+        /// <value><c>true</c> if usage of equality comparer for equals should be disabled; otherwise, <c>false</c>.</value>
+        protected bool DisableUsageOfEqualityComparerForEquals { get; set; }
```

Comment:
I am not really sure about this one. I think the properties on the ModelBase are already a bit much (I'd love to make it more simple actually). Is GetType() really that expensive? If so, we should at least cache it for ModelBase so we only have to retrieve the GetType for the other object.